### ISSUE TITLE
It takes alot of time to update\reinstall

### ISSUE THREAD

**Issue by Guy-Markman** [Glavin001/atom-beautify#1675]
# Description

It takes about 5 minutes to update the package everytime there is a little update.
Any other package takes about 10-30seconds

In the last update it got worst.
I waited for 15 and nothing happend, when I tried uninstalling and installing again it got stuck again.
Stopped it, tried updating other packages and everything was ok.

# Steps to Reproduce

1. Open CMD
2. Write apm update atom-beautify
3. Wait

# Input
apm update atom-beautify

when I gave up:
apm uninstall atom-beautify
apm install atom-beattify 

stuck again

# Expected
atom-beautify installed

# Actualy

Installing atom-beautify to C:\Users\Offer\.atom\packages
and nothing after that

# Debug

Here is a link to the `debug.md` Gist: <INSERT GIST HERE>

Can't find where to get it for apm install\update

# Checklist

I have:
- [X] Tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [X] Reloaded (or restarted) Atom to ensure it is not a caching issue
- [X] Searched through existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues
so I know this is not a duplicate issue
- [X] Filled out the Input, Expected, and Actual sections above or have edited/removed them in a way that fully describes the issue. 
- [] Generated debugging information by executing `Atom Beautify: Help Debug Editor` command in Atom and added link for `debug.md` Gist to this issue [I don't know how to do it for apm install\update]

**Comment by Glavin001** [comment]
Atom-Beautify has a lot of dependencies. I'm working on improving this in the future.

Might I ask what your Internet speed is? See http://www.speedtest.net/ for test. 15 minutes is a very long time.

Any install timing test I do personally would be heavily biased:
![image](https://cloud.githubusercontent.com/assets/1885333/26792221/1dfb7f44-49f0-11e7-92a7-a784e8213d50.png)

**Comment by szeck87** [comment]
I can report the same issue. My connection is 200 Mbps down and 78 up. It's been going for at least an hour. I don't see any changes to the files in C:\users\user\.atom\packages\atom-beautify, they're all from 5/18. I do see changes in .atom\.apm\atom-beautify along with other folders in .apm based on dependencies for atom-beautify, but still taking a really long time.

**Comment by Glavin001** [comment]
This may even be an [`apm`](https://github.com/atom/apm) bug. It sounds like `apm` is not installing the dependencies properly.

Also, if Atom-Beautify only makes a small change and no dependency change it should only update (small changes) instead of doing a completely fresh install. It sounds like `apm` is doing a fresh install, which is slow. 
Maybe it would be a good idea to report to https://github.com/atom/apm as well?

**Comment by szeck87** [comment]
Yea possible. The next update I'll do it through npm command to include verbose instead of using atom UI.

**Comment by szeck87** [comment]
Updating via apm command, I can see it making requests to URLs that ultimately are not updated. Why it's taking so long in this environment I can't answer.

**Comment by Glavin001** [comment]
@szeck87 I wonder if it would be appropriate to ask at  https://github.com/atom/apm or https://discuss.atom.io/ and see if anyone has any ideas? While I know Atom-Beautify does have a lot of dependencies, and I do plan on tackling this issue, it will likely be much longer before that release comes. In the mean time, there may be some optimizations within `apm` which can benefit Atom-Beautify users and more!

**Comment by szeck87** [comment]
Ok sounds like a plan, I'll open something with that project.

For tackling the dependency issue, can separating atom-beautify from the actual beautifiers be done? I typically only need it for the languages I code in, I don't need all.

**Comment by Glavin001** [comment]
> For tackling the dependency issue, can separating atom-beautify from the actual beautifiers be done? I typically only need it for the languages I code in, I don't need all.

Definitely on the roadmap. It was discussed in https://github.com/Glavin001/atom-beautify/issues/620 and other issues. It will be included in Unibeautify, which is a big undertaking and in progress. See https://github.com/Unibeautify/beautifier-prettydiff for example of a Unibeautify beautifier, which will be both a separate npm package and run as a separate beautifier service to Unibeautify core ( https://github.com/Unibeautify/unibeautify ).
Atom-Beautify will be powered by Unibeautify in the long run: https://github.com/Glavin001/atom-beautify/issues/1174

Lots of great enhancements coming! However, I am only 1 man and need time to finish it up 😄 .

**Comment by szeck87** [comment]
Updating to 0.30.2 from 0.30.1, it's spent the last two hours updating .cache.json files in registry.npmjs.org folder and package.json files in .apm/package_name/version_number/package folder for each dependency. So yea, its making http calls to the individual dependency packages for something.

**Comment by Glavin001** [comment]
Wow great research! I think this is definitely something the apm or npm team should know about. I'm really hoping there is something which could be done :crossed_fingers:

**Comment by Bobobrien92** [comment]
I still had this problem today.  I'm not sure how long the actual install takes but it seemed severely lagging for me still.  I searched for the package unibeautify and atom-beautify showed up the list of relevant packages as installed.  When i searched atom-beautify again it still showed as installed instead of installing.

**Comment by szeck87** [comment]
@Bobobrien92 are you behind a firewall/proxy?  This is the first update to Atom Beautify in a while so there were a lot of dependencies being updated at the same time.  I've found that when you're behind those firewalls it can really slow down installs/updates when npm is involved.

**Comment by Bobobrien92** [comment]
@szeck87 Yes I do have windows firewall on.  That could be it

**Comment by szeck87** [comment]
Sorry I meant a network firewall, though I suppose anti-virus software and Windows firewall _could_ have an effect.  Do you have an approximation of how long it took?  Was it more than 10 minutes, more than 30 minutes, an hour?

**Comment by Bobobrien92** [comment]
The lack of other recent comments makes me feel like this may be an issue on my part.  I waited about fifteen minutes the first time before I quit and relaunched Atom.  When I tried the beautify shortcut I got an error about pretty-diff not being installed.  I uninstalled Atom then tried reinstalling.  While waiting about five minutes I searched for a solution and saw one post saying check for updates after an appropriate amount of time then go back to search.  Mine was still installing.  Maybe it was trivial that I searched for a different package and afterwards found the package installed and rendered complete in the Atom UI
